airbyte-webapp/src/packages/cloud/cloudRoutes.tsx

```diff
@@ -40,21 +37,7 @@ const CloudSettingsPage = React.lazy(() => import("./views/settings/CloudSetting
 const DefaultView = React.lazy(() => import("./views/DefaultView"));
 
 const MainRoutes: React.FC = () => {
-  const { setWorkspaceFeatures } = useFeatureService();
   const workspace = useCurrentWorkspace();
-  const cloudWorkspace = useGetCloudWorkspace(workspace.workspaceId);
-
-  useEffect(() => {
-    const outOfCredits =
-      cloudWorkspace.creditStatus === CreditStatus.NEGATIVE_BEYOND_GRACE_PERIOD ||
-      cloudWorkspace.creditStatus === CreditStatus.NEGATIVE_MAX_THRESHOLD;
-    // If the workspace is out of credits it doesn't allow creation of new connections
-    // or syncing existing connections.
-    setWorkspaceFeatures(outOfCredits ? ({ [FeatureItem.AllowSync]: false } as FeatureSet) : []);
```

Comment:
Enforcing the policy by dynamically toggling an app-wide feature setting feels like a really hacky use of the feature system anyway; not sad to see it go.